cmd/vic-machine/create/create.go

```diff
@@ -874,16 +876,43 @@ func (c *Create) processDNSServers() error {
 
 func (c *Create) processVolumeStores() error {
 	defer trace.End(trace.Begin(""))
-	c.VolumeLocations = make(map[string]string)
+	c.VolumeLocations = make(map[string]*url.URL)
 	for _, arg := range c.volumeStores {
-		if err := common.CheckUnsupportedCharsDatastore(arg); err != nil {
-			return fmt.Errorf("--volume-store contains unsupported characters: %s Allowed characters are alphanumeric, space and symbols - _ ( ) / :", err)
+		splitMeta := strings.Split(arg, ":")
+		if len(splitMeta) < 2 {
+			return errors.New("volume store input must be in format datastore/path:label or ")
 		}
-		splitMeta := strings.SplitN(arg, ":", 2)
-		if len(splitMeta) != 2 {
-			return errors.New("Volume store input must be in format datastore/path:label")
+
+		// divide out the label with the target
+		lastIndex := len(splitMeta)
+		label := splitMeta[lastIndex-1]
+		rawTarget := strings.Join(splitMeta[0:lastIndex-1], ":")
+
+		// raw target input should be in the form of a url
+		urlTarget, err := url.Parse(rawTarget)
+		if err != nil {
+			return fmt.Errorf("parsed url for option --volume-store does not have scheme 'nfs',  valid inputs are datastore/path:label or %s", nfsInputformat)
```

Comment:
yeah that needs an upgrade. I should likely say something more along the lines of "invalid url input, ..." stating the proper input.